### ISSUE TITLE
allowCardToPlayWithoutMana doesn't seem to be requiring additional costs

### ISSUE THREAD

**Issue by Zerrisx** [magefree/mage#6760]
Idol of Endurance and Narset, Enlightened Master both use this routine, and cards cast off their abilities ignore any additional costs.  Headless Skaab doesn't require you to exile a creature from your graveyard, Lesser Manticore doesn't require you to discard a card, and Altar's Reap doesn't require you to sacrifice a creature.

**Comment by 18ths** [comment]
Can confirm this and additionally throw some additional info. Seems like we have a test for this showing that this interaction works correctly, suggesting that we could have a bug in test framework too. See org.mage.test.cards.asthough.PlayFromNonHandZoneTest.testNarsetEnlightenedMasterAdditionalCost()

**Comment by JayDi85** [comment]
I tested it by tests and by manual game test in current dev version -- all works fine. `Narset, Enlightened Master` asks 2 cards to discard from `Cathartic Reunion`

**Comment by 18ths** [comment]
I'm also testing it right now and still having the same issues - seems like AI tries to cast the spell as soon as its exiled. For some reason the cost is then set as paid, despite ai having no creatures to sacrifice (Im using Altar's Reap), which is why you dont have to pay the cost, as its already paid.

I'll try to update master branch, maybe it was just fixed somehow. My top commit now is currently 74ceac7322d7fe4994778c870e692ba9c122c106

**Comment by JayDi85** [comment]
It's game simulations (screenshot). You are missed one of the test command and AI try to solve it.

Use that commands all the time:
* `setStrictChooseMode(true);`
* `assertAllCommandsUsed();`

**Comment by 18ths** [comment]
I wasn't using automated tests, this was an actual game.

**Comment by 18ths** [comment]
Still broken on current master. My init file:

```
[init]
battlefield:Human:Forest:5
battlefield:Human:Plains:5
battlefield:Human:Mountain:5
battlefield:Human:Swamp:5
battlefield:Human:Island:5

[narset]
@init
library:Human:Altar's Reap:1
battlefield:Human:Narset, Enlightened Master:1
battlefield:Human:Concordant Crossroads:1
```

**Comment by JayDi85** [comment]
Must be fixed by 802aca942a8da10b3fcb71c08a11868011865dcc

It's missing cost copy (AI take payed status from last simulation instead new pay).

**Comment by 18ths** [comment]
Yup, that was it. Thanks, I'd spend a decade trying to figure it out. ;)
